docs/content/develop/make-a-breaking-change.md

```diff
@@ -1,211 +1,213 @@
+2. Add deprecations and warnings to the `main` branch of `magic-modules`
+3. Add upgrade guide entries to the `main` branch of `magic-modules`
+4. Make the breaking change on `FEATURE-BRANCH-major-release-{{% param "majorVersion" %}}`
+
+These are covered in more detail in the following sections. The upgrade guide
+and the actual breaking change will be merged only after both are completed.
+
+### Make the `main` branch forwards-compatible with the major release
+
+What forwards-compatibility means will vary depending on the breaking change. For example:
+
+* If a required field is being removed, make the field optional
+  on the `main` branch.
+* If a field is being renamed, the new field must be added to the `main` branch
+
+
+### Add deprecations and warnings to the `main` branch of `magic-modules`
+
+Deprecations and warnings must be actionable at the time that they are added
+to the `main` branch, and they must be added prior to the `{{% param "majorVersion" %}}`
+major release. Every deprecation or warning should be surfaced to users of the
+provider at runtime as well as in documentation.
+
+#### Field deprecation (due to removal or rename)
+
+{{< tabs "Field deprecations" >}}
+{{< tab "MMv1" >}}
+Set `deprecation_message` on the field. For example:
+
+```yaml
+- !ruby/object:Api::Type::String
+  name: 'apiFieldName'
+  description: |
+    MULTILINE_FIELD_DESCRIPTION
+  deprecation_message: api_field_name is deprecated and will be removed in a future major release. Use other_field_name instead.
+```
+
+The deprecation message will automatically show up in the API documentation.
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+1. Set `Deprecated` on the field. For example:
+
+   ```go
+   "api_field_name": {
+      Type:       schema.String,
+      Deprecated: "api_field_name is deprecated and will be removed in a future major release. Use other_field_name instead.",
+      ...
+   }
+   ```
+2. Update the [documentation for the field]({{< ref "/develop/resource#add-documentation" >}}) to include the deprecation notice. For example:
+
+   ```markdown
+   * `api_field_name` - (Optional, [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html)) FIELD_DESCRIPTION. `api_field_name` is deprecated and will be removed in a future major release. Use `other_field_name` instead.
+   ```
+{{< /tab >}}
+{{< /tabs >}}
+
+#### Resource deprecation (due to removal or rename)
+
+{{< tabs "Resource deprecations" >}}
+{{< tab "MMv1" >}}
+1. Set `docs.warning` and `custom_code.resource_definition` on the resource. For example:
+
+   ```yaml
+   docs: !ruby/object:Provider::Terraform::Docs
+     warning: |
+       google_RESOURCE_NAME is deprecated and will be removed in the next major release
+       of the provider. Use google_OTHER_RESOURCE_NAME instead.
+   custom_code: !ruby/object:Provider::Terraform::CustomCode
+     resource_definition: templates/terraform/resource_definition/RESOURCE_NAME_deprecation.go.erb
```

Comment:
Is the resourc_definition necessary? What does it do?

Reply:
Oh, I see. I think recently we've added `deprecation_message` as an attribute. That should replace this step.

Reply:
I don't think we support it yet so I've created a PR to add support at https://github.com/GoogleCloudPlatform/magic-modules/pull/8696. I'll update this PR accordingly.

Reply:
done in fd875dd